ChangeLog

```diff
@@ -80,6 +80,11 @@ Release date: TBA
   Closes #6069
   Closes #6136
 
+* ``lru-cache-decorating-method`` has been renamed to ``cache-max-size-none`` and
```

Comment:
`maxsize` is spelled as one word in the argument. What about `cache-maxsize-none`?

Reply:
If we keep it like `max-size` we can potentially reuse later on for another function. But I'm fine with either!

Reply:
I don't feel strongly. Happy to keep this way.